specification/trace/semantic_conventions/http.md

```diff
@@ -153,6 +153,13 @@ This span type represents an outbound HTTP request. There are two ways this can
 
 For an HTTP client span, `SpanKind` MUST be `Client`.
 
+HTTP client spans SHOULD end sometime after all the response headers are received. This MAY or MAY NOT include time spent reading the response body.
+
+HTTP client library instrumentations SHOULD document their behavior around ending HTTP client spans.
+
+> **Note**
+> It is nice for HTTP client spans to end once the response body has been fully read. However, if there is any possibility for application code to not fully read the HTTP response (and for the HTTP client library to then have to clean up the HTTP response asynchronously), the HTTP client span SHOULD be ended earlier (e.g. after the HTTP response headers are read). This avoids having to end the span asynchronously later on at a time which is no longer directly associated with the application code which made the HTTP request.
```

Comment:
> However, if there is any possibility for application code to not fully read the HTTP response (and for the HTTP client library to then have to clean up the HTTP response asynchronously)

Someone writing an HTTP client or adding telemetry to it can't know how the client will be used by an app. There is the potential for any HTTP request to be started, not used, and then timeout asynchronously. This could happen before or after receiving response headers.

Reply:
I added

> e.g. after the HTTP response headers are read, **or fail to be read**

let me know if that sounds better, thx

Reply:
No, the problem remains. The paragraph can be reduced down to: "this information is better to capture, but you should only do it in this situation which will never happen".

> Someone writing an HTTP client or adding telemetry to it can't know how an app will use the client.

To go into more detail, these counters are always located inside general-purpose HTTP clients:
* HttpClient for .NET
* OkHttp for Java
* XHR in Javascript
* etc

General-purpose HTTP clients don't know with certainty how the library or app will use HTTP requests. They'll all give guidance that someone must close/dispose an HTTP request once it's no longer needed, but they can't guarantee that someone will do that.

Reply:
> General-purpose HTTP clients don't know with certainty how the library or app will use HTTP requests

great, I believe we are all in agreement on this

> The paragraph can be reduced down to: "this information is better to capture, but you should only do it in this situation which will never happen"

HTTP semantic conventions also covers other cases outside of general-purpose HTTP client instrumentation